test.sh

```diff
@@ -221,4 +240,6 @@ prepare_testdir_under_gopath
 
 generate_crd_resources
 test_generated_controller
+run_dep_ensure
+test_generated_controller
```

Comment:
I don't expect `test_generated_controller` to be called twice. Maybe you only need it after `run_dep_ensure`.

Reply:
I wanted to keep the tests as close as possible to the end user's workflow. users or we don't invoke `dep ensure` currently, so wanted to test controller tests before and after `dep ensure`.

I will add a comment explaining that. We will eventually have test controller when we start running `dep ensure` in project setup itself.

Reply:
That's an interesting angle I hadn't thought of! That makes a lot more sense now.

Reply:
Added a comment in script explaining the reasoning.